### PR TITLE
Ping pong with error

### LINKED ISSUE #255: Complete IBC Mock testing
This builds on what is done in #253 and adds some simpler demos as defined in our milestone:

Add a mockHandler that plays ping-pong:
  * version is "ping" or "pong" (both acceptable)
  * on version negotiation the "Try" side will switch the version (if "ping" requested, counterparty does "pong" and visa versa)
  * stores a counter in state (starts at 0)
  * Packets are JSON, only valid ones look like `{"ping": 5}`
  * Receiver will add the value to the local state and return acknowledgement
  * Acknowledgement is either `{"error": "message here"}` or `{"success": {"pong": 14}}` or whatever the updated state is
  * On receiving acknowledgement, if a success, then that "remoteState" is stored. If error, increment "numErrors"

This handler has "localState" and "remoteState". Receiving a "ping" packet will increment "localState". Getting a "pong" acknowlegdement will set "remoteState".

Once this is build, a few scenarios:
  * Proper connection (show version is swapped from original request), sending packets with coordinators, it works well
  * Send `{"random": "data"}` packet and verify we get an `"error"` acknowledgement back (numErrors == 1)
  * Try connecting with invalid version (transfer) and show channel will not get established

### PR DESCRIPTION
**Pull request by alpe**
Resolves #255 

Add tests for unhappy path

### CODE REVIEW

File: x/wasm/relay_pingpong_test.go [hunk 1]
```diff
@@ -121,6 +120,189 @@ func TestPinPong(t *testing.T) {
...
+		activePlayer  = ping
+		pingBallValue = startValue
+	)
+	for i := 1; i <= rounds-1; i++ { // play some rounds before reaching max value
+		t.Logf("++ round: %d\n", i)
+		ball := NewHit(activePlayer, pingBallValue)
+
+		seq := uint64(i)
+		pkg := channeltypes.NewPacket(ball.GetBytes(), seq, channelA.PortID, channelA.ID, channelB.PortID, channelB.ID, doNotTimeout, 0)
+		ack := ball.BuildAck()
```
Comment (ethanfrey) [inline x/wasm/relay_pingpong_test.go:181]: I just noticed this. The NewPacket creates and sends a packet, right, but it doesn't return the ack bytes?
You manually create the ack in the test code (not from the contract).

Is this due to unclear APIs? Is it not possible to read the packet/ack from the chain?

File: x/wasm/relay_pingpong_test.go [hunk 2]
```diff
@@ -121,6 +120,189 @@ func TestPinPong(t *testing.T) {
...
+
+		err = coordinator.RelayPacket(chainA, chainB, clientA, clientB, pkg, ack.GetBytes())
```
Comment (ethanfrey) [inline x/wasm/relay_pingpong_test.go:183]: This is odd, you pass in the ack instead of it coming from the contract

**Review by ethanfrey — APPROVED** [review]
Good test cases, but some comments on where the packet/ack come from?

File: x/wasm/relay_pingpong_test.go (hunk 2 above)
Comment (alpe) [reply x/wasm/relay_pingpong_test.go:183]: This is a "workaround" in ibc tests. The relayer should watch the chain instead and read the packet/ack information from the events exposed there. Unfortunately this is not implemented into the coordinator (yet) and also not a simple task that I could squeeze in easily.
This setup is also used in other ibc tests so we are in good company at least.

File: x/wasm/relay_pingpong_test.go (hunk 1 above)
Comment (alpe) [reply x/wasm/relay_pingpong_test.go:181]: `channeltypes.NewPacket` does not sent a packet here. It only creates an object that is used below with the coordinator to be passed to the other chain (B). This object creation would not be necessary in a full relayer demo. See my other comment.

File: x/wasm/relay_pingpong_test.go (hunk 2 above)
Comment (ethanfrey) [reply x/wasm/relay_pingpong_test.go:183]: Fair enough, limits of test environment.

**Pull request merged**